cirq/circuits/circuit.py

```diff
@@ -1258,6 +1258,60 @@ def _json_dict_(self):
     def _from_json_dict_(cls, moments, device, **kwargs):
         return cls(moments, strategy=InsertStrategy.EARLIEST, device=device)
 
+    def raggedy_add(*circuits: 'cirq.Circuit') -> 'cirq.Circuit':
```

Comment:
Should this be a module-level method since it doesn't contain a self or cls reference?

Reply:
Also would it work with `AbstractCircuit` input?

Reply:
This way of defining the method makes it usable as both an instance method and a class method, while preserving the ability to pass in any number of arguments. `c1.raggedy_add(c2)` works, `cirq.Circuit.raggedy_add(c1, c2)` works, `cirq.Circuit.raggedy_add()` works, `cirq.Circuit.raggedy_add(c1, c2, c3)` works, `c1.raggedy_add(c2, c3, c4)` works, etc.

Reply:
The reason I mention it is there's a roadmap item to simplify circuit construction at #3246. Some of the discussion in linked issue #2775 is that perhaps Circuit itself should only contain the core insert/append, whereas the more specialized functions could live as module-level methods, maybe in a separate builder module eventually, so that Circuit class itself remains lightweight. Once something is added to Circuit proper, it's hard to remove.

Reply:
(And my second comment was in reference to the signature: should the parameter instead be `*circuits: 'cirq.AbstractCircuit'`)?

Reply:
Done. Parameter type loosened, and added to FrozenCircuit also.